interfaces/builtin/qualcomm_ipc_router.go

```diff
@@ -40,32 +45,232 @@ const qipcrtrBaseDeclarationSlots = `
+`
+
+const qipcrtrConnectedPlugAppArmorCompat = `
+# Description: allows access to the Qualcomm IPC Router sockets
+#              and limits to sock_dgram only
+network qipcrtr,
+
+# CAP_NET_ADMIN required for port number smaller QRTR_MIN_EPH_SOCKET per 'https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/net/qrtr/qrtr.c'
+capability net_admin,
 `
 
 const qipcrtrConnectedPlugAppArmor = `
 # Description: allows access to the Qualcomm IPC Router sockets
 #              and limits to sock_dgram only
 network qipcrtr,
 
+unix (connect, send, receive) type=seqpacket addr="###SOCKET_ADDRESS###" peer=(label=###SLOT_SECURITY_TAGS###),
+`
+
+const qipcrtrPermanentSlotAppArmor = `
+# Description: allows access to the Qualcomm IPC Router sockets
+#              and limits to sock_dgram only
+network qipcrtr,
+
 # CAP_NET_ADMIN required for port number smaller QRTR_MIN_EPH_SOCKET per 'https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/net/qrtr/qrtr.c'
 capability net_admin,
+capability net_bind_service,
+
+unix (bind, listen) type=seqpacket addr="###SOCKET_ADDRESS###",
+`
+
+const qipcrtrConnectedSlotAppArmor = `
+unix (accept, send, receive) type=seqpacket addr="###SOCKET_ADDRESS###" peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const qipcrtrConnectedPlugSecCompCompat = `
+# Description: allows access to the Qualcomm IPC Router sockets
+bind
+
+# We allow AF_QIPCRTR in the default template since it is mediated via the AppArmor rule
+#socket AF_QIPCRTR
 `
 
-const qipcrtrConnectedPlugSecComp = `
+const qipcrtrPermanentSlotSecComp = `
 # Description: allows access to the Qualcomm IPC Router sockets
 bind
+accept
+accept4
+listen
 
 # We allow AF_QIPCRTR in the default template since it is mediated via the AppArmor rule
 #socket AF_QIPCRTR
 `
 
+const (
+	// qcipcAttrib is a tag that must match between plug and slot
+	qcipcAttrib = "qcipc"
+	// addressAttrib is the allowed socket address and is specified in the slot
+	addressAttrib = "address"
+)
+
 type qualcomIPCRouterInterface struct {
 	commonInterface
 }
 
-func (iface *qualcomIPCRouterInterface) BeforeConnectPlug(plug *interfaces.ConnectedPlug) error {
+func (iface *qualcomIPCRouterInterface) Name() string {
+	return "qualcomm-ipc-router"
+}
+
+func (iface *qualcomIPCRouterInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              qipcrtrSummary,
+		BaseDeclarationSlots: qipcrtrBaseDeclarationSlots,
+		// backward compatibility
+		ImplicitOnCore: true,
+		// backward compatibility
+		ImplicitOnClassic: true,
+	}
+}
+
+func fillSnippetSocketAddress(slot *interfaces.ConnectedSlot, snippet string) (string, error) {
+	var address string
+	if err := slot.Attr(addressAttrib, &address); err != nil {
+		return "", err
+	}
+
+	return strings.ReplaceAll(snippet, "###SOCKET_ADDRESS###", address), nil
+}
+
+// Note that we cannot use implicitSystemConnectedSlot as legacy slots can be
+// for UC too in this case.
```

Comment:
we have a PR to change that now:

https://github.com/snapcore/snapd/pull/13194

maybe we should leave a XXX or TODO here, to drop this when that lands

Reply:
Done